### ISSUE TITLE
Make the artifact create endpoint idempotent

### ISSUE THREAD

**Issue by daviddavis** [pulp/pulpcore#5780]
The package creation endpoints are idempotent so I wonder if the artifact create endpoint should be as well. Currently, you get a 400 error if you attempt to upload an artifact that already exists:

```
$ https --form :/pulp/api/v3/artifacts/ file@~/frigg_1.0_ppc64.deb
{
    "non_field_errors": [
        "Artifact with sha256 checksum of 'a20723d4d174abecc0001bbbe183d04fc26b2ca30d77d87b5c4c9f3be35b67ee' already exists."
    ]
}
```

**Comment by quba42** [comment]
Have other plugins already solved this? In other words is there already an API endpoint somewhere for creating a multi-artifact content without having to separately create the artifacts first? That is how I understood the request.

**Comment by dkliban** [comment]
I don't think we have an example of this anywhere else. @dralley @ggainey does RPM have anything like this?

**Comment by quba42** [comment]
So I just had a look at the pulp_deb Source Package API endpoint, and it expects you to provide the `.dsc` file as an artifact, and then it will just loop over the files referenced in that `.dsc` file, expecting those to already exist as artifacts in Pulp.

Does anyone have any ideas how one might provide a content API endpoint with multiple uploads to create a content and multiple artifacts from?

I did not think about this before, and if it does not already exist elsewhere, I am not at all confident I know a good approach. :smile:

**Comment by mdellweg** [comment]
I think the multipart body can accept multiple files in one request just fine. Not sure if DRF covers for that.
Also I would kind of expect that for each file you can either upload a file, specify an upload or (for the time being) an artifact.

Somehow we are derailing from the original ask: "Should the Artifact upload be idempotent?" And we _can_ do that.
